python/ray/data/_internal/numpy_support.py

```diff
@@ -62,11 +69,11 @@ def convert_udf_returns_to_numpy(udf_return_col: Any) -> Any:
         # `str` are also Iterable.
         try:
             # Try to cast the inner scalars to numpy as well, to avoid unnecessarily
-            # creating an inefficient array of array of object dtype. Don't convert
-            # scalar lists though, since those can be represented as pyarrow list type
-            # without needing to go through our tensor extension.
-            if all(
-                is_valid_udf_return(e) and not is_scalar_list(e) for e in udf_return_col
```

Comment:
Don't think `is_scalar_list` is used anywhere other than here. Let's remove the function?